### PR TITLE
31 implement company repository

### PR DESCRIPTION
**Pull request by davidlefrancq**
This pull request introduces an in-memory repository for managing company data, complete with robust error handling and comprehensive unit tests. The main changes include seeding company data, implementing the `CompanyRepository` with standard CRUD operations, defining custom error classes, and adding tests to ensure repository reliability.

**Repository and Data Management:**

* Added a new `CompanyRepository` class that manages an in-memory list of companies, supporting listing all companies, retrieving by ID, and partial updates with validation and error handling.
* Introduced a seed data file `companies.seed.json` containing sample company records for use in the repository.

**Error Handling:**

* Created custom error classes (`GetAllCompaniesError`, `GetCompanyByIdError`, `UpdateCompanyError`) to provide clear and specific error messages for repository operations.

**Testing:**

* Added a comprehensive test suite for `CompanyRepository`, covering successful operations, error cases, and edge conditions to ensure robust behavior.

### CODE REVIEW

File: src/repositories/errors/CompanyRepositoryError.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+export class GetCompanyByIdError extends CompanyRepositoryError {
+  constructor() {
+    super('Failed to retrieve company by id.');
+    this.name = 'GetCompanyByIdError';
+  }
+}
+
+export class UpdateCompanyError extends CompanyRepositoryError {
+  constructor() {
+    super('Failed to update company.');
```
[the remark this answers is not in the document]
Comment (davidlefrancq) [reply src/repositories/errors/CompanyRepositoryError.ts:24]: Not suitable for setting up a translation system.

File: src/repositories/__tests__/CompanyRepository.spec.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    await expect(repo.updatePartial(validId, {})).rejects.toThrow(UpdateCompanyError);
+  });
+
+  it('list() throws GetAllCompaniesError on internal error', async () => {
+    // Simulate internal corruption
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    repo.companies = null;
+    await expect(repo.list()).rejects.toThrow(GetAllCompaniesError);
+  });
```
[the remark this answers is not in the document]
Comment (davidlefrancq) [reply src/repositories/__tests__/CompanyRepository.spec.ts:86]: It's specific test with bad typing.

**Review by DevLfq — APPROVED** [review]
